plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/delete/EqualityDeleteFilter.java

```diff
@@ -47,22 +45,27 @@ public RowPredicate createPredicate(List<IcebergColumnHandle> columns)
                 .toArray(Type[]::new);
 
         Schema fileSchema = schemaFromHandles(columns);
-        StructProjection projection = StructProjection.create(fileSchema, schema);
+        equalityDeleteSets.initialize(fileSchema);
 
         return (page, position) -> {
             StructLike row = new LazyTrinoRow(types, page, position);
-            return !deleteSet.contains(projection.wrap(row));
+            return !equalityDeleteSets.contains(row);
         };
     }
 
-    public static DeleteFilter readEqualityDeletes(ConnectorPageSource pageSource, List<IcebergColumnHandle> columns, Schema tableSchema)
+    public static void readEqualityDeletes(
+            ConnectorPageSource pageSource,
+            List<IcebergColumnHandle> columns,
+            Set<Integer> ids,
+            EqualityDeleteSets equalityDeleteSets,
```

Comment:
Rather than passing in both `ids` and `equalityDeleteSets` here, can we just pass in the correct `StructLikeSet deleteSet` to add these entries to?

Reply:
I thought about doing this before, but was worried about moving all the logic into the IcebergPageSourceProvider, if you have no comments, I will modify it.